ovs/lib/setup.py

```diff
@@ -1201,35 +1415,47 @@ def _demote_node(cluster_ip, master_ip, cluster_name, ip_client_map, unique_id,
         print 'Leaving arakoon ovsdb cluster'
         logger.info('Leaving arakoon ovsdb cluster')
         offline_node_ips = [node.ip for node in offline_nodes]
-        ArakoonInstaller.shrink_cluster(cluster_ip, 'ovsdb', offline_node_ips)
+        try:
+            ArakoonInstaller.shrink_cluster(cluster_ip, 'ovsdb', offline_node_ips)
+        except Exception as ex:
+            SetupController._print_log_error('leave arakoon ovsdb cluster', ex)
```

Comment:
Any idea why that might fail? It might be a better idea to make ```shrink_cluster``` smarter so it checks reality and just assures that once the method completes, the state is as expected (so that if you try to shrink a cluster twice, the second time it won't do anything)